libs/mutantschemata/cpp_source/rewrite.hpp

```diff
@@ -0,0 +1,794 @@
+                    break;
+                case clang::BO_NE:
+                    insertMutantSchemata(binOp, {clang::BO_EQ}, {Singleton::True});
+                    break;
+                default:
+                    break;
+            }
+        } else {
+            //TODO verify this it's ok to use general rules and hope for the best
+            switch (binOp->getOpcode()) {
+                    // Relational operators
+                case clang::BO_LT:
+                    insertMutantSchemata(binOp, {clang::BO_LE, clang::BO_NE}, {Singleton::False});
+                    break;
+                case clang::BO_GT:
+                    insertMutantSchemata(binOp, {clang::BO_GE, clang::BO_NE}, {Singleton::False});
+                    break;
+                case clang::BO_LE:
+                    insertMutantSchemata(binOp, {clang::BO_LT, clang::BO_EQ}, {Singleton::True});
+                    break;
+                case clang::BO_GE:
+                    insertMutantSchemata(binOp, {clang::BO_GT, clang::BO_EQ}, {Singleton::True});
+                    break;
+                    // Equality operators
+                case clang::BO_EQ:
+                    insertMutantSchemata(binOp, {clang::BO_LE, clang::BO_GE}, {Singleton::False});
+                    break;
+                case clang::BO_NE:
+                    insertMutantSchemata(binOp, {clang::BO_LT, clang::BO_GT}, {Singleton::True});
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+    
+    
+    void mutateBinaryAOR(clang::BinaryOperator *binOp) {
+        switch (binOp->getOpcode()) {
+                // Additive operators
+            case clang::BO_Add:
+                insertMutantSchemata(binOp, {clang::BO_Sub, clang::BO_Mul, clang::BO_Div, clang::BO_Rem}, {Singleton::LHS, Singleton::RHS});
+                break;
+            case clang::BO_Sub:
+                insertMutantSchemata(binOp, {clang::BO_Add, clang::BO_Mul, clang::BO_Div, clang::BO_Rem}, {Singleton::LHS, Singleton::RHS});
+                break;
+                // Multiplicative operators
+            case clang::BO_Mul:
+                insertMutantSchemata(binOp, {clang::BO_Sub, clang::BO_Add, clang::BO_Div, clang::BO_Rem}, {Singleton::LHS, Singleton::RHS});
+                break;
+            case clang::BO_Div:
+                insertMutantSchemata(binOp, {clang::BO_Sub, clang::BO_Mul, clang::BO_Add, clang::BO_Rem}, {Singleton::LHS, Singleton::RHS});
+                break;
+            case clang::BO_Rem:
+                insertMutantSchemata(binOp, {clang::BO_Sub, clang::BO_Mul, clang::BO_Div, clang::BO_Add}, {Singleton::LHS, Singleton::RHS});
+                break;
+            default:
+                break;
+        }
+    }
+    
+    
+    void mutateBinaryLCR(clang::BinaryOperator *binOp) {
+        switch (binOp->getOpcode()) {
+                // Logical operators
```

Comment:
Instead of these comments, maybe write methods with names like handle_logical_operators and call them with suitable arguments